### PR TITLE
import lua@5.3 from homebrew/versions, rename lua51 to lua@5.1, and remove universal builds from lua*

### PR DESCRIPTION
**Pull request by lembacon**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/lua@5.3.rb [hunk 1]
```diff
@@ -0,0 +1,169 @@
+class LuaAT53 < Formula
+  desc "Powerful, lightweight programming language"
+  homepage "https://www.lua.org/"
+  url "https://www.lua.org/ftp/lua-5.3.3.tar.gz"
+  sha256 "5113c06884f7de453ce57702abaac1d618307f33f6789fa870e87a59d772aca2"
+
+  option "with-default-names", "Don't version-suffix the Lua installation. Conflicts with Homebrew's Lua"
```
Comment (MikeMcQuaid) [inline Formula/lua@5.3.rb]: Let's remove this option, thanks.

File: Formula/lua@5.3.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  option "without-luarocks", "Don't build with Luarocks support embedded"
+
+  # Be sure to build a dylib, or else runtime modules will pull in another static copy of liblua = crashy
+  # See: https://github.com/Homebrew/homebrew/pull/5043
+  # ***Update me with each version bump!***
+  patch :DATA
```
Comment (MikeMcQuaid) [inline Formula/lua@5.3.rb]: Reckon there's any way we can use `inreplace` or otherwise avoid this patch?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
@lembacon Nice work! Would also be great to rename `lua51` accordingly, too, and make a PR in homebrew/versions to migrate this formula there too.

File: Formula/lua@5.3.rb (hunk 2 above)
Comment (lembacon) [inline Formula/lua@5.3.rb]: @MikeMcQuaid This patch hack can be found in formulae for lua 5.1, 5.2 & 5.3. Perhaps let's just leave this for a future pull request?

**Comment by lembacon** [comment]
@MikeMcQuaid 

- `--with-default-names` has been removed from `lua@5.3`.
- `lua51` has been renamed to `lua@5.1`, `--universal` has been removed and some minor cleanups have been applied in order to match `lua@5.3`.
- A pull request Homebrew/homebrew-versions#1515 has been submitted to migrate `lua53`.

**Comment by lembacon** [comment]
- `--universal` has been removed from `lua`.
- Formulae that depend on `lua51` have been updated to depend on `lua@5.1` with revision bumps.

**Comment by DomT4** [comment]
FWIW, I'd done most of this locally already, but the new `luarocks` update which needs to be applied seems to have "issues" with accepting suffixed `lua` executables. But at some point that will need doing. Not a blocker here, just an FYI.

File: Formula/radare2.rb [hunk 3]
```diff
@@ -24,6 +24,7 @@ class Radare2 < Formula
   stable do
     url "http://cloud.radare.org/get/1.1.0/radare2-1.1.0.tar.gz"
     sha256 "7bc1e206a2b4def6bdb8684c2af0281b007986a0b5b5da652bd03be264ca0fa5"
+    revision 1
```
Comment (DomT4) [inline Formula/radare2.rb]: This is in the wrong place. Applies to the whole formula, so needs to live outside any `stable`, etc block.

**Comment by lembacon** [comment]
The misplaced `revision` in `radare2.rb` has been fixed. 🙈

File: Formula/lua@5.3.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/lua@5.3.rb]: Sure thing. If you're ever able to look at that it'd be ❤️

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request closed without merging**